lib/finch.ex

```diff
@@ -536,4 +542,9 @@ defmodule Finch do
   defp get_pool(%Request{scheme: scheme, host: host, port: port}, name) do
     PoolManager.get_pool(name, {scheme, host, port})
   end
+
+  def get_pool_status(finch_name, shp) do
```

Comment:
Maybe we should also allow the user to pass the url as a string that we would parse into a shp just as a convenience function? We could mention in the docs that it is more efficient to pass the shp if that is of concern.